src/main/java/com/facebook/openwifirrm/ucentral/models/AggregatedState.java

```diff
@@ -0,0 +1,234 @@
+		}
+	}
+
+	public String bssid;
+	public String station;
+	public long connected;
+	public long inactive;
+	public List<Integer> rssi;
+	public long rxBytes;
+	public long rxPackets;
+	public AggregatedRate rxRate;
+	public long txBytes;
+	public long txDuration;
+	public long txFailed;
+	public long txPackets;
+	public AggregatedRate txRate;
+	public long txRetries;
+	public int ackSignal;
+	public int ackSignalAvg;
+	public Radio radio;
+
+	public static class Radio {
+		public int channel;
+		public int channelWidth;
+		public int txPower;
+
+		public Radio() {}
+
+		public Radio(int channel, int channelWidth, int txPower) {
+			this.channel = channel;
+			this.channelWidth = channelWidth;
+			this.txPower = txPower;
+		}
+
+		public Radio(Map<String, Integer> radioInfo) {
+			channel = radioInfo.getOrDefault("channel", null);
+			channelWidth = radioInfo.getOrDefault("channel_width", null);
+			txPower = radioInfo.getOrDefault("tx_power", null);
+		}
+
+		@Override
+		public int hashCode() {
+			return Objects.hash(channel, channelWidth, txPower);
+		}
+
+		@Override
+		public boolean equals(Object obj) {
+			if (this == obj) {
+				return true;
+			}
+			if (obj == null) {
+				return false;
+			}
+			if (getClass() != obj.getClass()) {
+				return false;
+			}
+
+			Radio other = (Radio) obj;
+			return channel == other.channel &&
+				channelWidth == other.channelWidth &&
+				txPower == other.txPower;
+		}
+	}
+
+	@Override
+	public int hashCode() {
+		return Objects.hash(bssid, station, radio);
+	}
+
+	@Override
+	public boolean equals(Object obj) {
+		if (this == obj) {
+			return true;
+		}
+		if (obj == null) {
+			return false;
+		}
+		if (getClass() != obj.getClass()) {
+			return false;
+		}
+
+		AggregatedState other = (AggregatedState) obj;
+
+		return bssid == other.bssid &&
+			station == other.station &&
+			connected == other.connected && inactive == other.inactive && rssi
+				.equals(other.rssi) &&
+			rxBytes == other.rxBytes && rxBytes == other.rxPackets &&
+			rxRate.bitRate == other.rxRate.bitRate &&
+			rxRate.chWidth == other.rxRate.chWidth && rxRate.mcs
+				.equals(other.rxRate.mcs) &&
+			txBytes == other.txBytes && txDuration == other.txDuration &&
+			txFailed == other.txFailed && txPackets == other.txPackets &&
+			txRate.bitRate == other.txRate.bitRate &&
+			txRate.chWidth == other.txRate.chWidth && txRate.mcs
+				.equals(other.txRate.mcs) &&
+			txRetries == other.txRetries && ackSignal == other.ackSignal &&
+			ackSignalAvg == other.ackSignalAvg && radio.equals(other.radio);
+	}
+
+	/** Constructor with no args */
+	public AggregatedState() {
+		this.rxRate = new AggregatedRate();
+		this.txRate = new AggregatedRate();
+		this.rssi = new ArrayList<>();
+		this.radio = new Radio();
+	}
+
+	/** Construct from Aggregatedstate */
+	public AggregatedState(AggregatedState state) {
+		this.rxRate = new AggregatedRate();
+		this.txRate = new AggregatedRate();
+		this.rssi = new ArrayList<>();
+		add(state);
```

Comment:
Doesn't it always return false because `state.hashCode()` is different from `hashCode(null,null,null)`?

Reply:
Maybe I need to add one more condition check in add(state)